### PR TITLE
ICU-22371 Fix ULocale.getISOLanguages() API docs

### PR DESCRIPTION
**Pull request by gvictor**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22371
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [ ] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: icu4j/main/classes/core/src/com/ibm/icu/util/ULocale.java [hunk 1]
```diff
@@ -877,8 +877,9 @@ public static String[] getISOCountries() {
     }
 
     /**
-     * Returns a list of all 2-letter language codes defined in ISO 639.
-     * Can be used to create Locales.
+     * Returns a list of all 2-letter language codes defined in ISO 639, plus additional
```
Comment (mihnita) [inline icu4j/main/classes/core/src/com/ibm/icu/util/ULocale.java:880]: Sorry for the delay, I didn't see it.

I think I would stay away from "determined to be useful" and just mention the facts (what is returned) and point to BCP 47.

What do you think about:
```
Returns a list of all unique language codes defined in ISO 639. They can be 2 or 3 letter codes, as defined by [BCP 47, section 2.2.1](https://www.ietf.org/rfc/bcp/bcp47.html#section-2.2.1).
```

Comment (gvictor) [reply icu4j/main/classes/core/src/com/ibm/icu/util/ULocale.java]: Done

**Comment by gvictor** [comment]
Mihai, could you help review this CL again?

**Review by mihnita — APPROVED** [review]
Thank you,
M.

**Review by markusicu — APPROVED** [review]
lgtm

**Pull request merged**